tools/generator/src/Generator/XCSchemeInfo+TestActionInfo.swift

```diff
@@ -101,8 +101,16 @@ Expected at least one target in `TestAction.targets`
         }
         for testActionTargetId in testActionTargetIds {
             if let testActionTargetEnv: [String: String] = envs[testActionTargetId] {
-                env.merge(testActionTargetEnv) { current, _ in
-                    current
+                for (key, value) in testActionTargetEnv {
+                    if let existingValue: String = env[key] {
+                        let errorMessage: String = """
+                        Detected conflicting environment variables:
```

Comment:
@brentleyjones is this the right approach for conflicts?

Reply:
Should be a `UserError`, as `PreconditionError` is for bugs in the generator.

We only want to error if the values differ, because if they are the same they can merge just fine. I would use `.merging()` as you can throw in there by simply checking if the values differ.